ddapm_test_agent/agent.py

```diff
@@ -652,6 +642,64 @@ async def handle_trace_analyze(self, request: Request) -> web.Response:
         # wait 1s, gather traces and assert tags
         raise NotImplementedError
 
+    @middleware  # type: ignore
+    async def store_request_middleware(self, request: Request, handler: _Handler) -> web.Response:
+        # only store requests for specific endpoints
+        if request.path in self._forward_endpoints:
+            await self._store_request(request)
+
+        headers = CIMultiDict(request.headers)
+
+        if "X-Datadog-Trace-Env-Variables" in headers:
+            var_string = headers.pop("X-Datadog-Trace-Env-Variables")
+            env_vars = {
+                key.strip(): value.strip() for key, value in (pair.split("=") for pair in var_string.split(","))
+            }
+            request["_dd_trace_env_variables"] = env_vars
+
+        if "X-Datadog-Agent-Proxy-Disabled" in headers:
+            request["_proxy_to_agent"] = headers.pop("X-Datadog-Agent-Proxy-Disabled").lower() != "true"
```

Comment:
this check should be in the proxy codepath, not the request storing

Reply:
IMO I don't agree. I think centrally locating any header transformations / checks is useful. To get the same functionality as before, I would have to do this header check within each of the forward endpoints (traces, stats, telemetry).  I added the check to make sure agent_url isn't empty. Should be the same functionality as before.

Reply:
wouldn't it be centralized in the proxy codepath/middleware?